sqlglot/dialects.py

```diff
@@ -328,6 +337,12 @@ def _ts_or_ds_to_date_sql(self, expression):
         this = self.sql(expression, "this")
         return f"DATE_PARSE(SUBSTR({this}, 1, 10), '%Y-%m-%d')"
 
+    def _ts_or_ds_add_sql(self, expression):
+        this = self.sql(expression, "this")
+        e = self.sql(expression, "expression")
+        unit = self.sql(expression, "unit") or "'day'"
+        return f"DATE_FORMAT(DATE_ADD({unit}, {e}, {this}), '%Y-%m-%d')"
```

Comment:
this is presto? it doesn't look quite right because e is a ts or ds, wouldn't it fail?+

Reply:
`e` would be a number no?
this is a ts or ds

Reply:
presto add is
```
        exp.DateAdd: lambda self, e: f"""DATE_ADD({self.sql(e, 'unit') or "'day'"}, {self.sql(e, 'expression')}, {self.sql(e, 'this')})""",
```

Reply:
nno, ts or ds is time string or date string

DateAdd in presto expects a date type, not a string type

Reply:
my bad, {this} is a date or a timestamp
{e} is a number

Reply:
i think this is still wrong